src/bufr/IodaEncoder/IodaDescription.h

```diff
@@ -41,8 +44,52 @@ namespace Ingester
         int compressionLevel;  // Optional
     };
 
+    struct GlobalDescriptionBase
+    {
+        std::string name;
+        int vectorSize;
+        virtual void addTo(ioda::Group& group) = 0;
+    };
+
+    template<typename T>
+    struct is_vector : public std::false_type {};
+
+    template<typename T, typename A>
+    struct is_vector<std::vector<T, A>> : public std::true_type {};
+
+    template<typename T>
+    struct GlobalDescription : public GlobalDescriptionBase
+    {
+        T value;
+
+        void addTo(ioda::Group& group) final
+        {
+            _addTo(group);
+        }
+
+     private:
+        // T is something other than a std::vector
+        template<typename U = void>
+        void _addTo(ioda::Group& group,
+                    std::enable_if_t<!is_vector<T>::value, U>* = nullptr)
+        {
+            ioda::Attribute attr = group.atts.create<T>(name, {1});
+            attr.write<T>({value});
+        }
+
+        // T is a vector
+        template<typename U = void>
+        void _addTo(ioda::Group& group,
+                    std::enable_if_t<is_vector<T>::value, U>* = nullptr)
+        {
+            ioda::Attribute attr = group.atts.create<typename T::value_type>(name, {vectorSize});
```

Comment:
Instead of `{vectorSize}` you can just call `value.size()` (just assume the value is a vector, the compiler will let you know if you made a mistake).

Reply:
I tried this but it doesn't work. The compiler will complain something.

Reply:
Let me know how it goes.

Reply:
This is what I got: 
```
/data/users/nchen/CodeSprintDec20/src/ioda-bundle/iodaconv/src/bufr/IodaEncoder/IodaDescription.h(85): error #3291: invalid narrowing conversion from "unsigned long" to "long"
              ioda::Attribute attr = group.atts.create<typename T::value_type>(name, {value.size()});
```

Reply:
I found if I use int(value.size()) then it will compile. Don't quite understand why I need to do so but I'll eliminate that extra variable then.

Reply:
So the type returned by value.size() is a size_t which is actually an "unsigned long". The problem is that the function group.atts.create takes a "long" which is a little surprising actually :) So instead of int(value.size()) the more correct thing would be to do static_cast<long>(value.size()). Thanks for figuring this out!

Reply:
Yeah I used ``` static_cast<int>(value.size())``` and passed all the test. I think a regular int should be good enough for specifying the length of this vector in our case.